connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java

```diff
@@ -0,0 +1,226 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.util.clusters;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.kafka.connect.runtime.Connect;
+import org.apache.kafka.connect.runtime.Herder;
+import org.apache.kafka.connect.runtime.Worker;
+import org.apache.kafka.connect.runtime.WorkerConfigTransformer;
+import org.apache.kafka.connect.runtime.distributed.DistributedConfig;
+import org.apache.kafka.connect.runtime.distributed.DistributedHerder;
+import org.apache.kafka.connect.runtime.isolation.Plugins;
+import org.apache.kafka.connect.runtime.rest.RestServer;
+import org.apache.kafka.connect.storage.ConfigBackingStore;
+import org.apache.kafka.connect.storage.Converter;
+import org.apache.kafka.connect.storage.KafkaConfigBackingStore;
+import org.apache.kafka.connect.storage.KafkaOffsetBackingStore;
+import org.apache.kafka.connect.storage.KafkaStatusBackingStore;
+import org.apache.kafka.connect.storage.StatusBackingStore;
+import org.apache.kafka.connect.util.ConnectUtils;
+import org.junit.rules.ExternalResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+public class EmbeddedConnectCluster extends ExternalResource {
+
+    private static final Logger log = LoggerFactory.getLogger(EmbeddedConnectCluster.class);
+
+    private static final int NUM_BROKERS = 1;
+    private static final Properties BROKER_CONFIG = new Properties();
+
+    private final EmbeddedKafkaCluster kafkaCluster = new EmbeddedKafkaCluster(NUM_BROKERS, BROKER_CONFIG);
+
+    private final Map<String, String> workerProps;
+    private final String clusterName;
+
+    private Connect connect;
+    private URI advertisedUrl;
+
+    public EmbeddedConnectCluster(Class<?> klass) {
+        this(klass.getName());
+    }
+
+    public EmbeddedConnectCluster(String name) {
+        // this empty map will be populated with defaults before starting Connect.
+        this(name, new HashMap<>());
+    }
+
+    public EmbeddedConnectCluster(String name, Map<String, String> workerProps) {
+        this.workerProps = workerProps;
+        this.clusterName = name;
+    }
+
+    public void start() throws IOException {
+        before();
+    }
+
+    /**
+     * Stop the connect cluster and the embedded Kafka and
+     */
+    public void stop() {
+        after();
+    }
+
+    @Override
+    protected void before() throws IOException {
+        kafkaCluster.before();
+        startConnect();
+        log.info("Started connect at {} with kafka cluster at {}", restUrl(), kafka().bootstrapServers());
```

Comment:
Kafka, ZooKeeper use upper case already. 
`Connect` will be easier not to confuse with the verb `connect` even if the context helps